pkg/controller/statusmanager/pod_status.go

```diff
@@ -84,53 +84,61 @@ func SetNodeCondition(conditions *[]corev1.NodeCondition, newCondition corev1.No
 		*conditions = append(*conditions, newCondition)
 		return true
 	}
-
 	if existingCondition.Status != newCondition.Status {
 		existingCondition.Status = newCondition.Status
-		existingCondition.LastTransitionTime = metav1.NewTime(time.Now())
 		changed = true
 	}
-
-	existingCondition.Reason = newCondition.Reason
-	existingCondition.Message = newCondition.Message
+	if existingCondition.Reason != newCondition.Reason {
+		existingCondition.Reason = newCondition.Reason
+		changed = true
+	}
+	if existingCondition.Message != newCondition.Message {
+		existingCondition.Message = newCondition.Message
+		changed = true
+	}
+	if changed {
+		existingCondition.LastTransitionTime = metav1.NewTime(time.Now())
+	}
 	return changed
 }
 
-func (status *StatusManager) setNodeNetworkUnavailable(nodeName string, ready bool, reason string, message string, sharedInfo *sharedinfo.SharedInfo) {
+func (status *StatusManager) setNodeNetworkUnavailable(nodeName string, ready bool, startedAt *time.Time, reason string, message string, sharedInfo *sharedinfo.SharedInfo) {
 	if sharedInfo.LastNetworkAvailable == nil {
 		sharedInfo.LastNetworkAvailable = make(map[string]time.Time)
 	}
+	if ready {
+		log.Info(fmt.Sprintf("Last network unavailable time of node %s is set to %s", nodeName, *startedAt))
+		sharedInfo.LastNetworkAvailable[nodeName] = *startedAt
+	} else {
+		delete(sharedInfo.LastNetworkAvailable, nodeName)
+	}
 	_, changed := status.combineNode(nodeName, ready, reason, message)
 	if !changed {
 		log.Info("Node condition is not changed")
 		return
 	}
 	log.Info(fmt.Sprintf("Setting status NetworkUnavailable to %v for node %s", !ready, nodeName))
 	if !ready {
-		delete(sharedInfo.LastNetworkAvailable, nodeName)
 		status.updateNodeStatus(nodeName, ready, reason, message)
 		return
 	}
-	last := time.Now()
-	_, present := sharedInfo.LastNetworkAvailable[nodeName]
-	if !present {
-		sharedInfo.LastNetworkAvailable[nodeName] = last
-		log.Info(fmt.Sprintf("Last network unavailable time of node %s is set to %s", nodeName, last))
-
-		go func() {
-			time.Sleep(time.Duration(operatortypes.TimeBeforeRecoverNetwork) * time.Second)
-			if (sharedInfo.LastNetworkAvailable[nodeName]) != last {
-				log.Info(fmt.Sprintf("Last network unavailable time of node %s changed, expecting %v, got %v, goroutine exiting",
-					nodeName, last, sharedInfo.LastNetworkAvailable[nodeName]))
-				return
-			}
-			status.updateNodeStatus(nodeName, ready, reason, message)
+	// When the network status looks ready, we still need to wait for a graceful time
+	now := time.Now()
+	passedSeconds := int64(now.Sub(*startedAt) / 1000000000)
```

Comment:
How about using time duration for `1000000000`

Reply:
Done